cpu/esp32/esp-ieee802154/esp_ieee802154_hal.c

```diff
@@ -0,0 +1,466 @@
+/*
+ * Copyright (C) 2025 Gunar Schorcht
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+#include "esp_ieee802154.h"
+
+#include "esp_ieee802154_hal.h"
+#include "log.h"
+#include "net/ieee802154/radio.h"
+
+#define ENABLE_DEBUG 0
+#include "debug.h"
+
+/* ACK frame timeout in microseconds, should be a multiple of 16 */
+#define ESP_IEEE802154_ACK_TIMEOUT_US (3456)
+
+_Static_assert((3456 % 16 == 0), "ACK frame timeout should be a multiple of 16");
+
+/* Although the device driver supports IEEE802154_CAP_IRQ_TX_START,
+ * IEEE802154_CAP_IRQ_RX_START and IEEE802154_CAP_IRQ_CCA_DONE, we are not
+ * using it for now to avoid unnecessary performance degradation as it is
+ * not used by any link layer driver */
+#define _USE_RX_START   0
+#define _USE_TX_START   0
+#define _USE_CCA_DONE   0
+
+#define _USE_SET_RX     1
+#define _USE_SET_IDLE   1
+
+static const ieee802154_radio_ops_t esp_ieee802154_driver;
+static ieee802154_dev_t *esp_ieee802154_dev;
+
+static uint8_t  _tx_frame[IEEE802154_FRAME_LEN_MAX + 1]; /* additional byte 0 is used for length */
+static uint8_t *_rx_frame;
+static const uint8_t *_ack_frame;
+
+static esp_ieee802154_frame_info_t *_rx_frame_info;
+static esp_ieee802154_frame_info_t *_ack_frame_info;
+
+static bool _channel_clear;
+
+static esp_ieee802154_tx_error_t _tx_error;     /* error on last transmit */
+
+static int _write(ieee802154_dev_t *dev, const iolist_t *psdu)
+{
+    (void)dev;
+
+    DEBUG("[esp_ieee802154] %s\n", __func__);
+
+    assert(psdu);
+
+    memset(_tx_frame, 0, sizeof(_tx_frame));
+
+    /* copy packet data into the _tx_frame buffer */
+    unsigned len = 0;
```

Comment:
```suggestion
    unsigned len = 1; /* first byte is frame length */
```
makes the code a bit easier to understand.

or simply 

```C
    uint8_t *dst = &_tx_frame[1];
```